### ISSUE TITLE
New Revised Ratings/Order Question

### ISSUE THREAD

**Issue by newman-iknewit** [SudokuMonster/SukakuExplainer#103]
In the following grid:

`+----------------------+----------------------+----------------------+
| 345678 45679  45789  | 567    6789   3569   | 267    279    1      |
| 5678   5679   1      | 567    6789   2      | 67     3      4      |
| 36     2      79     | 16     4      136    | 5      79     8      |
+----------------------+----------------------+----------------------+
| 458    1459   4589   | 2      19     149    | 3      6      7      |
| 46     1469   249    | 3      169    7      | 8      124    5      |
| 467    3      247    | 8      5      146    | 124    124    9      |
+----------------------+----------------------+----------------------+
| 1      457    6      | 9      3      8      | 47     457    2      |
| 2      457    3      | 1457   17     15     | 9      8      6      |
| 9      8      457    | 4567   2      56     | 147    1457   3      |
+----------------------+----------------------+----------------------+`

With the new ratings order this gets a 4.3, with 3 Turbot Fishes found (a 2 string kite, a grouped skyscraper, and a grouped 2 string kite). However, there is a Swordfish available on 6 (c257r125), which reduces it to singles. I'm assuming the new order always applies the turbot fish first, even if they are a higher rating? Is this intended?

**Comment by SudokuMonster** [comment]
Thanks,
The ordering of techniques is prioritized over rating. You are correct in you observation that in the new rating The order would be Turbot Fish Techniques followed by swordfish. The ratings of  Turbot Fish Techniques appear to be the same both in the classic order or the new order which can be rectified by lowering the Turbot fish techniques to make them lower than Swordfish only in that setup.

As you observed a lower rated Swordfish (until this is rectified) could be present in the same puzzle.

Effectively this shouldn't change the solving path (as it depends on Technique ordering rather than rating). You would in an ideal world want you ratings to parallel technique ordering though.

I'll highlight it as a bug for now & may change this in the next release!

Thanks again

**Comment by newman-iknewit** [comment]
Perhaps this could be a configuration option - whether to prioritize order of technique or rating when presented with overlapping ranges.

I'm sure there are lots of different views on which techniques are actually harder anyway. :)

**Comment by SudokuMonster** [comment]
With Turbot Fish techniques they _Should_ be easier than swordfish as they depend on 2 strong links in the logic which swordfish depends on 3.

SE within a technique will order & choose the elimination ... extending that to more one technique will impact speed of solving ultimate as SE can be used to bulk-rate puzzles.

As you said there different views out there.

SE has this issue (which is well known) with any 2 solving techniques that have overlapping rating. A prime example is Uniqueness (which can go beyond 5.0 in rating in some settings) and the Naked Quad.

**Comment by newman-iknewit** [comment]
Yeah, it's not going to make a huge difference either way. I do think some Swordfish (and Jellyfish) are much easier to spot than others as a manual solver, though that likely isn't the case for SE. For example:

[code]...........98....7.8..6..5..5..4..3...79....2...........27....9.4..5..6.3....62..[/code]

has immediate Swordfish on 2, 5, 6, 7, and 9 just from the (fishy) geometry of the givens. (A few of the eliminations on 2 and 6 can also be found with turbot fish, so this gets a 4.3 as well.)

I can definitely understand the impact it would have on bulk solving though, especially for harder puzzles.

**Comment by SudokuMonster** [comment]
Yes I know that trick ... It was used before to construct some special Pure Jellyfish puzzles 

[http://forum.enjoysudoku.com/a-pure-jellyfish-collection-t5776-79.html](url)

**Comment by newman-iknewit** [comment]
Yeah, the Jellyfish is arguably even easier to spot in those puzzles than the Swordfish here... if you know to look for it anyway. :)
